instrumentation/opentelemetry_oban/lib/opentelemetry_oban/job_handler.ex

```diff
@@ -0,0 +1,102 @@
+defmodule OpentelemetryOban.JobHandler do
+  alias OpenTelemetry.Span
+
+  @tracer_id :opentelemetry_oban
+
+  def attach(opts \\ []) do
+    attach_job_start_handler(opts)
+    attach_job_stop_handler(opts)
+    attach_job_exception_handler(opts)
+  end
+
+  defp attach_job_start_handler(opts) do
+    :telemetry.attach(
+      "#{__MODULE__}.job_start",
+      [:oban, :job, :start],
+      &__MODULE__.handle_job_start/4,
+      opts
+    )
+  end
+
+  defp attach_job_stop_handler(opts) do
+    :telemetry.attach(
+      "#{__MODULE__}.job_stop",
+      [:oban, :job, :stop],
+      &__MODULE__.handle_job_stop/4,
+      opts
+    )
+  end
+
+  defp attach_job_exception_handler(opts) do
+    :telemetry.attach(
+      "#{__MODULE__}.job_exception",
+      [:oban, :job, :exception],
+      &__MODULE__.handle_job_exception/4,
+      opts
+    )
+  end
+
+  def handle_job_start(_event, _measurements, metadata, sampler: sampler) do
+    %{
+      job: %{
+        id: id,
+        queue: queue,
+        worker: worker,
+        priority: priority,
+        inserted_at: inserted_at,
+        scheduled_at: scheduled_at,
+        attempt: attempt,
+        max_attempts: max_attempts,
+        meta: job_meta
+      }
+    } = metadata
+
+    :otel_propagator_text_map.extract(Map.to_list(job_meta))
+    parent = OpenTelemetry.Tracer.current_span_ctx()
+    links = if parent == :undefined, do: [], else: [OpenTelemetry.link(parent)]
```

Comment:
It may be worth noting in docs that linking is not supported by a lot of tooling, especially commercial tools.

Reply:
I believe Honeycomb does support this, I see nice links between the process and send spans :)